### PR TITLE
added architecture diagram

### PR DESCRIPTION
**Pull request by victorjarlow**
Draft at a architecture diagram for use when communicating features and functions in the ATOS ecosystem

### CODE REVIEW

**Comment by victorjarlow** [comment]
@RI-SE/night-team please provide some feedback on the figure. Is it clear enough? is it well structured? Is info missing? What improvements are needed?

**Comment by samuelthoren** [comment]
Looks good! I just have some minor questions or clarifications.
- Should there be more text in the EsminiAdapter box? It looks like the text suddenly ends.
- I might be mistaken here, but do the external systems use JournalControl?
- PointcloudPublisher publishes once on init, not periodically.
- Maybe you can specify what type of data is sent using the MQTT-bridge
- Specify what resetting test actors means in BackToStart.
- The isoObject-box is quite empty, maybe we can write some things there?
- The letter D in the legend, specify what type of dynamic input.

Otherwise it looks really good, nice overview of ATOS!

**Comment by victorjarlow** [comment]
> Looks good! I just have some minor questions or clarifications.
> 
> * Should there be more text in the EsminiAdapter box? It looks like the text suddenly ends.

Added!
> * I might be mistaken here, but do the external systems use JournalControl?

In my mind, the filesystem is an external system that gets test reports provided to it. I didn't want to explicitly list all data-sinks, but instead keep it "high-level"
> * PointcloudPublisher publishes once on init, not periodically.

Thanks for the clarification!
> * Maybe you can specify what type of data is sent using the MQTT-bridge

To avoid having to change this diagram constantly, I think it's better to keep it high-level.
> * Specify what resetting test actors means in BackToStart.

Done
> * The isoObject-box is quite empty, maybe we can write some things there?

Added some text
> * The letter D in the legend, specify what type of dynamic input.

I'm not sure how I could describe the dynamic data in such a way that it fits all three modules (esminiAdapter - triggers and actions, DirectControl - steering signals, TrajectoryletStreamer - trajectory chunks).
I don't want to introduce too much detail in this diagram such that we have to modify it constantly. Also, I think that a  user could go to our docs page for a specific module if they want to know in more detail.

**Comment by Robert108** [comment]
Nice to have a diagram like this! I would put the image in the res folder in the docs folder and link it from there.
I think it could be nice to write just a few words also on how you should go about updating this diagram if you need to.

File: docs/ATOS_architecture [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+            <mxPoint x="800" y="500" as="targetPoint" />
+            <Array as="points">
+              <mxPoint x="440" y="240" />
+              <mxPoint x="530" y="240" />
+              <mxPoint x="530" y="460" />
+              <mxPoint x="800" y="460" />
+            </Array>
+          </mxGeometry>
+        </mxCell>
+        <mxCell id="Nz_zJu77zt_nOGFol6SB-70" value="&lt;b&gt;PointcloudPublisher&lt;/b&gt;&lt;br&gt;Provides pointcloud vizualization of the test-environment" style="rounded=0;whiteSpace=wrap;html=1;" parent="1" vertex="1">
```
Comment (samuelthoren) [inline docs/ATOS_architecture]:
```suggestion
        <mxCell id="Nz_zJu77zt_nOGFol6SB-70" value="&lt;b&gt;PointcloudPublisher&lt;/b&gt;&lt;br&gt;Provides pointcloud visualization of the test-environment" style="rounded=0;whiteSpace=wrap;html=1;" parent="1" vertex="1">
```

**Review by samuelthoren — APPROVED** [review]

**Comment by samuelthoren** [comment]
Can I merge this @victorjarlow?

**Comment by victorjarlow** [comment]
> Can I merge this @victorjarlow?

I've not moved the document as Robert suggested yet..

**Pull request merged**